src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -0,0 +1,45 @@
+-module(mongoose_system_metrics_collector).
+
+-type report_struct() :: 
+    #{
+        report_name := term(),
+        key := term(),
+        value := term()
+    }.
+
+-export_type([report_struct/0]).
+
+-export([collect/1]).
+
+collect(ClientId) ->
+    ReportResults = [ get_reports(RGetter) || RGetter <- report_getters()],
+    FlatReportResults = lists:flatten(ReportResults),
+    spawn(mongoose_system_metrics_sender, send, [ClientId, FlatReportResults]).
```

Comment:
Maybe this doesn't really need to be spawned, this process will die as another one is created, and it doesn't really make any performance or scalability improvement at all to continue the work outside of it 🤷‍♂